stdlib/public/SwiftShims/Visibility.h

```diff
@@ -18,6 +18,10 @@
 #ifndef SWIFT_STDLIB_SHIMS_VISIBILITY_H
 #define SWIFT_STDLIB_SHIMS_VISIBILITY_H
 
+#if !defined(__GNUC__)
```

Comment:
You can do this in general using `!defined(__has_feature)`, as seen on the very next line. Clang treats it as defined. (And GCC doesn't have __has_feature either AFAIK.)